### PR TITLE
Center it

### PR DESCRIPTION
**Pull request by rderimay**
Fixed the rotation algorithm to work also when image as been cropped
and rotated
The focus boxes are rotated accordingly on the view
Made some refactoring, in particular :
- Special case for Ricoh cameras has been removed as unnecessary now
(see next point)
- DefaultDelegates.metaKeyAfPointUsed is now a table allowing for
multiple possibilities to be tried until one is found
- All function returning angles should now return it in trigonometric
sense

### CODE REVIEW

**Comment by rderimay** [comment]
Rotation example :-)

capture 2017-01-05 a 20 33 14
capture 2017-01-05 a 20 33 28

File: focuspoints.lrdevplugin/DefaultDelegates.lua [hunk 1]
```diff
@@ -55,19 +56,22 @@ end
...
+  -- returns 90, 270, or 0 (in trigonometric sense)
 --]]
 function DefaultDelegates.getShotOrientation(photo, metaData)
   local dimens = photo:getFormattedMetadata("dimensions")
   local orgPhotoW, orgPhotoH = parseDimens(dimens) -- original dimension before any cropping
 
   local metaOrientation = ExifUtils.findValue(metaData, "Orientation")
-  if (string.match(metaOrientation, "90") and orgPhotoW < orgPhotoH) then
+  if (string.match(metaOrientation, "90 CW") or string.match(metaOrientation, "90")) and orgPhotoW < orgPhotoH then
+    return 270
```
Comment (musselwhizzle) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: why is this backwards? A metadata of "Rotate 90 CW" used to return a value of 90. and "Rotate 270 CW" used to return 270.

Also ```string.match(metaOrientation, "90 CW")``` is a duplicate of  ```string.match(metaOrientation, "90")``` "90" will match anything with a "90" in it.

**Comment by musselwhizzle** [comment]
Looks great man. Let's tidy up the rotation stuff, I'll PR into the branch, run some tests on my local machine and then ship it.

**Comment by rderimay** [comment]
I wrote it in the commit I think this is to have all functions return angles in trigonometric sense and not sometime so and sometimes otherwise

File: focuspoints.lrdevplugin/DefaultDelegates.lua (hunk 1 above)
Comment (rderimay) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: correct. Should have used .find for example. You want me to commit the correction ?

Comment (musselwhizzle) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: yes pls

**Comment by rderimay** [comment]
done

**Comment by musselwhizzle** [comment]
If you'd like to show up as a "contributor", do an insignificant PR into master. It doesn't seem to give you credit if i PR into a branch. Your code has been merged into master.

**Comment by rderimay** [comment]
Done :-)


On 5 janvier 2017 at 22:33:39, Joshua Musselwhite (notifications@github.com) wrote:

If you'd like to show up as a "contributor", do an insignificant PR into master. It doesn't seem to give you credit if i PR into a branch. Your code has been merged into master.

**Comment by rderimay** [comment]
Tomorrow i will work on improving the code to be able to display af boxes of variable sizes that will open the way to more flexibility for different types of af (extended middle point, faces) etc. 
Time for a sleep now. 1:45 am here in France. 
Where are you from ?

**Comment by musselwhizzle** [comment]
very late there. I'm in Los Angeles. Thanks for all your work on this :)

**Comment by rderimay** [comment]
You're welcome. Nice, practical, project to start to code LR plugins in lua :-)
I will start a major rewrite of the whole af points draw code today, so it's better to concentrate on other stuff for now.
We should be able to support most of Canon and Fuji cameras at once after that. Did not look at the Nikons but that should be the same.


On 6 janvier 2017 at 02:02:57, Joshua Musselwhite (notifications@github.com) wrote:

very late there. I'm in Los Angeles. Thanks for all your work on this :)

**Comment by musselwhizzle** [comment]
May I ask that rendering/draw code be left as-is for now? I'm rather fond of the way it's structured and coded. As-is, it should be able to handle anything thrown at it. If you're excited to work on things, how about: 
1. Making an "Plug-in Author Tools" on page 31 of the manual so a section for logging. The users could dynamically toggle on and off the logger. 
2. Extract all the strings into a string files and get it ready for i18n. You could add French or any languages you know. 
3. Give Phil Harvey credit in the Plugin-in Author Tools (page 31) 
4. Continue helping with windows support.

**Comment by rderimay** [comment]
That's a bit too late, as I am almost finished with recoding everything and adding support to canon cameras without config files (only by using exif)
Also I made the display of everything much better and better structured anyway, including support of multiple af points and af point types as well as dynamic and correct sizing of them.
so...


On 6 janvier 2017 at 17:39:07, Joshua Musselwhite (notifications@github.com) wrote:

May I ask that rendering/draw code be left as-is for now? I'm rather fond of the way it's structured and coded. As-is, it should be able to handle anything thrown at it. If you're excited to work on things, how about:

Making an "Plug-in Author Tools" on page 31 of the manual so a section for logging. The users could dynamically toggle on and off the logger.
Extract all the strings into a string files and get it ready for i18n. You could add French or any languages you know.
Give Phil Harvey credit in the Plugin-in Author Tools (page 31)
Continue helping with windows support.

**Comment by musselwhizzle** [comment]
The code already allowed for any points to be dynamically defined without a mapping file. I'll put your PR into a branch.

**Comment by rderimay** [comment]
Could you send me a few Olympus files so I can see if everything is working correctly after adding the new olympus support back to my code please ?
After that I'm finished. This is really good imo. 

thx


On 6 janvier 2017 at 17:59:30, Joshua Musselwhite (notifications@github.com) wrote:

The code already allowed for any points to be dynamically defined without a mapping file. I'll put your PR into a branch.

**Comment by rderimay** [comment]
Sorry, forget the previous message, I thought you added Olympus support, I have contacted the guy who added it directly


On 6 janvier 2017 at 17:59:30, Joshua Musselwhite (notifications@github.com) wrote:

The code already allowed for any points to be dynamically defined without a mapping file. I'll put your PR into a branch.

**Comment by rderimay** [comment]
Ok. 
Af code has been PR. If could merge everything in the master before others start to write incompatible stuff. 
Tomorrow i can work on i18n for example.

**Comment by musselwhizzle** [comment]
I'm going to ask that you slow down on work. It's still my product that I have ideas and a direction that I want to take it. The latest PR, while at a glance looks nice, it huge and could have been broken into smaller units: 1 for dynamic canon's; 1 for different focus point sizes; etc. These are also things that can be discussed before hand. As the PR stands, it's something that will take me a while to review and test. I'll move it to a branch in the mean time. The most pressing issue seems is getting it to work on windows. I'm about at the point of give up and its seems LR is buggy on windows. If you got an VM, feel free to download a free trial and windows is all you :)

**Comment by musselwhizzle** [comment]
p.s. merged the code into feature/new_ideas. Some cool stuff there. Thanks for all the hard work.

**Pull request merged**